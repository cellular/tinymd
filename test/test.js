import tinymd from '../src';

const spec = (name, md, opts) =>
  test(name, () => {
    expect(tinymd(md, opts)).toMatchSnapshot();
  });

spec('Paragraph', 'Lorem ipsum');

spec(
  'Inline styles',
  '__bold__ regular ~~del~~ *italic* __bold__ ~~*deli*~~ **_boldi_** regular *ital*'
);

spec('Non-italic snake case', 'Text _italic_ snake_case_text');

spec(
  'Multiple paragraphs',
  `**bold**
line 1
line 2

line 3
**bold**
`
);

spec(
  'Code blocks',
  `Lorem
ipsum

    let md = 'hello';
    md += 'world';

dolor sit amed
`
);

spec(
  'Headings',
  `# h1
## h2
### h3
#### h4
##### h4
###### h6
####### no heading

#nospace-noheading

# trailing hashes #######################
# `
);

spec(
  'Unordered lists',
  `
* 1st level
   * 2nd level
   * 2nd
 * 2nd level
* 1st level`
);

spec(
  'Unorderd lists with dashes',
  `
* 1st level
   - 2nd level
   - 2nd
   - 2nd level
* 1st level`
);

spec(
  'Ordered lists',
  `
* 1st level
   1. 2nd level
   2. 2nd
   3. 2nd level
* 1st level`
);

spec(
  'Mixed lists',
  `
* 1st level
   1. 2nd level
   2. 2nd
   3. 2nd level
----`
);

spec(
  'Links',
  `lorem [example](http://example.com) ![alt](/image.jpg) [with title](/link.html "title") [blank target](+http://example.com) ipsum`
);

spec('Links with whitespace', `lorem [example]( mailto: me@example.com )`);

spec(
  'Horizontal rulers',
  `---
-- - - - - -
------------------------------------`
);

spec('Escaped asterisks', `aaa \\*bbb\\* ccc* \\\\ddd`);

spec('Escaped hashes', `aaa \\#bbb\\+ ccc\\- ddd`);

spec('Escaped chars', `aaa \\.bbb\\! ccc- ddd`);

spec('Escaped links', `aaa \\[bbb\\]\\(http://example.com\\) ccc ddd`);

spec('Escaped braces', `aaa \\{bbb\\} ccc`);

spec(
  'README example',
  `
# tinymd

supports ...

* lists
  - nested lists
    1. ordered
    2. with [links](http://example.com)
  - ![images](/cat.jpg)
  - rulers

----

All ~~common~~ **inline** _styles_

    code blocks
    rulers

And \\[escaping]\\(of special chars).
`
);

spec('Link rewriting', '[Issue 42](#42)', {
  rewrite: s => {
    const m = /^#(\d+)/.exec(s);
    return m ? `/issue${m[1]}` : s;
  }
});

spec('Header ids', '# Hello World', { addIds: true });

spec('Custom target _blank', '[foo](foo) [bar](bar)', {
  isBlank: s => s == 'foo'
});
