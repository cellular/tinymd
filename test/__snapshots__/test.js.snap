// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code blocks 1`] = `
"<p>Lorem
ipsum</p>

<pre><code>let md = 'hello';
md += 'world';</code></pre>

<p>dolor sit amed</p>
"
`;

exports[`Custom target _blank 1`] = `"<p><a href=\\"foo\\" target=\\"_blank\\">foo</a> <a href=\\"bar\\">bar</a></p>"`;

exports[`Escaped asterisks 1`] = `"<p>aaa *bbb* ccc* \\\\ddd</p>"`;

exports[`Escaped braces 1`] = `"<p>aaa {bbb} ccc</p>"`;

exports[`Escaped chars 1`] = `"<p>aaa .bbb! ccc- ddd</p>"`;

exports[`Escaped hashes 1`] = `"<p>aaa #bbb+ ccc- ddd</p>"`;

exports[`Escaped links 1`] = `"<p>aaa [bbb](http://example.com) ccc ddd</p>"`;

exports[`Header ids 1`] = `"<h1 id=\\"hello_world\\">Hello World</h1>"`;

exports[`Headings 1`] = `
"<h1>h1</h1><h2>h2</h2><h3>h3</h3><h4>h4</h4><h5>h4</h5><h6>h6</h6><p>####### no heading</p>

<p>#nospace-noheading</p>

<h1>trailing hashes</h1><h1></h1>"
`;

exports[`Horizontal rulers 1`] = `"<hr/><hr/><hr/>"`;

exports[`Images 1`] = `"<p>Hello <img src=\\"/cat.gif\\" alt=\\"\\"/></p>"`;

exports[`Inline styles 1`] = `"<p><b>bold</b> regular <del>del</del> <i>italic</i> <b>bold</b> <del><i>deli</i></del> <b><i>boldi</i></b> regular <i>ital</i></p>"`;

exports[`Link rewriting 1`] = `"<p><a href=\\"/issue42\\">Issue 42</a></p>"`;

exports[`Links 1`] = `"<p>lorem <a href=\\"http://example.com\\">example</a> <img src=\\"/image.jpg\\" alt=\\"alt\\"/> <a href=\\"/link.html\\" title=\\"title\\">with title</a> <a href=\\"http://example.com\\" target=\\"_blank\\">blank target</a> ipsum</p>"`;

exports[`Links with whitespace 1`] = `"<p>lorem <a href=\\"mailto: me@example.com\\">example</a></p>"`;

exports[`Mixed lists 1`] = `
"
<ul><li>1st level<ol><li>2nd level</li><li>2nd</li><li>2nd level</li></ol></li></ul><hr/>"
`;

exports[`Multiple paragraphs 1`] = `
"<p><b>bold</b>
line 1
line 2</p>

<p>line 3
<b>bold</b></p>
"
`;

exports[`Non-italic snake case 1`] = `"<p>Text <i>italic</i> snake_case_text</p>"`;

exports[`Ordered lists 1`] = `
"
<ul><li>1st level<ol><li>2nd level</li><li>2nd</li><li>2nd level</li></ol></li><li>1st level</li></ul>"
`;

exports[`Paragraph 1`] = `"<p>Lorem ipsum</p>"`;

exports[`README example 1`] = `
"
<h1>tinymd</h1>

<p>supports ...</p>

<ul><li>lists<ul><li>nested lists<ol><li>ordered</li><li>with <a href=\\"http://example.com\\">links</a></li></ol></li><li><img src=\\"/cat.jpg\\" alt=\\"images\\"/></li><li>rulers</li></ul></li></ul>

<hr/>

<p>All <del>common</del> <b>inline</b> <i>styles</i></p>

<pre><code>code blocks
rulers</code></pre>

<p>And [escaping](of special chars).</p>
"
`;

exports[`Unorderd lists with dashes 1`] = `
"
<ul><li>1st level<ul><li>2nd level</li><li>2nd</li><li>2nd level</li></ul></li><li>1st level</li></ul>"
`;

exports[`Unordered lists 1`] = `
"
<ul><li>1st level<ul><li>2nd level</li><li>2nd</li><li>2nd level</li></ul></li><li>1st level</li></ul>"
`;
